src/Akeneo/Pim/Enrichment/Bundle/Doctrine/ORM/Repository/ProductRepository.php

```diff
@@ -23,19 +24,29 @@ class ProductRepository extends EntityRepository implements
     IdentifiableObjectRepositoryInterface,
     CursorableRepositoryInterface
 {
+    private ?int $mainIdentifierId = null;
+
     /**
      * {@inheritdoc}
      */
     public function getItemsFromIdentifiers(array $identifiers)
     {
-        $qb = $this->createQueryBuilder('p')
-            ->where('p.identifier IN (:identifiers)')
-            ->setParameter('identifiers', $identifiers);
-
-        $query = $qb->getQuery();
-        $query->useQueryCache(false);
-
-        return $query->execute();
+        $uuids = $this->getEntityManager()->getConnection()->fetchFirstColumn(
+            <<<SQL
+            SELECT BIN_TO_UUID(product_uuid)
+            FROM pim_catalog_product_unique_data
+            WHERE attribute_id = :attributeId
+            AND raw_data IN (:identifiers)
+            SQL,
+            [
+                'identifiers' => $identifiers,
+                'attributeId' => $this->getMainIdentifierId(),
+            ], [
+                'identifiers' => Connection::PARAM_STR_ARRAY,
```

Comment:
```suggestion
                'identifiers' => Connection::PARAM_STR_ARRAY,
                'attributeId' => ParameterType::INTEGER,
```
(probably better for using the index)